pkg/controller.v1/tensorflow/pod.go

```diff
@@ -196,8 +196,7 @@ func (tc *TFController) createNewPod(tfjob *tfv1.TFJob, rt, index string, spec *
 		if podTemplate.Annotations == nil {
 			podTemplate.Annotations = map[string]string{}
 		}
-		// we create the podGroup with the same name as the tfjob
-		podTemplate.Annotations["scheduling.k8s.io/group-name"] = tfjob.Name
+		podTemplate.Annotations["scheduling.k8s.io/group-name"] = jobcontroller.GenPodGroupName(tfjob.Name)
```

Comment:
The annotation should be restricted to used with `kube-batch` scheduler?

Reply:
The annotation will be added if tc.Config.EnableGangScheduling is true, does it work for you?

Ref https://github.com/kubeflow/tf-operator/pull/1034/files/cb95e1629a47039c9ede83d96b904bc3cf17516d#diff-9f8bdf54c6d3e87e9087c9d7a9b804dcL187

Reply:
It depends on if the annotation is only defined by `kube-batch`? The other gang scheduler may not feel happy with it.

Reply:
AFAIK, our flag EnableGangScheduling is only designed for kube-batchd. And all the logic in this condition block is the logic for using kube-batchd. Thus I think we could keep it now.

If there are some other schedulers that can be used in tf-operator, we should have a design proposal to support different schedulers, IMO. And at that time, we should refactor the logic here.

WDYT @tossmilestone

Reply:
BTW, is there any competitor for kube-batchd now? I am not aware of it.

Reply:
If that's designated only for `kube-batch` for now, I think that's OK.

Reply:
Thanks.

And thanks for your review. Feel free to open an issue if you think we should support other gang schedulers.

:beers:

Reply:
🍺

Reply:
> BTW, is there any competitor for kube-batchd now? I am not aware of it.

Yes, that's a kubernetes project  `kubernetes-sigs/poseidon` also support gang scheudling.

Reply:
Oh I knew the project and its paper. Really awesome work. While none of us is familiar with the implementation now. I will take a look and try to support it if I can.

Thanks for your information.